bricks/test_optimizer/hooks/lib/pre_gen.dart

```diff
@@ -28,21 +28,55 @@ Future<void> run(HookContext context) async {
   final flutterSdkRegExp = RegExp(r'sdk:\s*flutter$', multiLine: true);
   final isFlutter = flutterSdkRegExp.hasMatch(pubspecContents);
 
-  final tests = testDir
-      .listSync(recursive: true)
-      .where((entity) => entity.isTest)
-      .map(
-        (entity) => path
-            .relative(entity.path, from: testDir.path)
-            .replaceAll(r'\', '/'),
-      )
-      .toList();
-
-  context.vars = {'tests': tests, 'isFlutter': isFlutter};
+  final identifierGenerator = StringIdGenerator();
+  final testIdentifierTable = <Map<String, String>>[];
+  for (final entity
+      in testDir.listSync(recursive: true).where((entity) => entity.isTest)) {
+    final relativePath =
+        path.relative(entity.path, from: testDir.path).replaceAll(r'\', '/');
+    testIdentifierTable.add({
+      'path': relativePath,
+      'identifier': identifierGenerator.next(),
+    });
+  }
+
+  context.vars = {'tests': testIdentifierTable, 'isFlutter': isFlutter};
 }
 
 extension on FileSystemEntity {
   bool get isTest {
     return this is File && path.basename(this.path).endsWith('_test.dart');
   }
 }
+
+/// {@template string_id_generator}
+/// A class that generates short identifiers.
+/// {@endtemplate}
+class StringIdGenerator {
```

Comment:
This algorithm was given by @wolfenrain .

If we want to adjust the logic of how identifier names are generated I'm open to changes.

Reply:
Side note:

Going to alter the algorithm to be a1, a2, a3, and so on. Since the current algorithm can generate words that are dart keywords which can't be used as indentifiers.